projects/igniteui-angular/src/lib/snackbar/snackbar.component.ts

```diff
@@ -214,28 +190,56 @@ export class IgxSnackbarComponent {
      * @hidden
      * @internal
      */
-    public snackbarMessage = '';
+    public snackbarMessage: string | OverlaySettings = '';
 
     /**
      * @hidden
      */
     private timeoutId;
 
-    constructor() { }
-
     /**
      * Shows the snackbar and hides it after the `displayTime` is over if `autoHide` is set to `true`.
      * ```typescript
      * this.snackbar.open();
      * ```
      */
-    public open(message?: string): void {
+    public open(message?: string | OverlaySettings) {
```

Comment:
Why do we need this union type? It doesn't make sense for the messsage argument to be of type OverlaySettings.